src/component/routers/GroupsRouter.test.tsx

```diff
@@ -0,0 +1,80 @@
+import * as React from 'react';
+import { FlowDefinition, SwitchRouter } from '../../flowTypes';
+import { composeSetup } from '../../utils';
+import { GROUP_LABEL } from './constants';
+import { extractGroups, GroupsRouter, GroupsRouterProps, hasGroupsRouter } from './GroupsRouter';
+import { GROUP_NOT_FOUND, GROUP_PLACEHOLDER } from '../form/GroupsElement';
+
+const config = require('../../../assets/config');
+const colorsFlow = require('../../../assets/flows/a4f64f1b-85bc-477e-b706-de313a022979.json');
+
+const definition = colorsFlow.results[0].definition as FlowDefinition;
+const { nodes: [sendMsgNode, , , , , groupsRouterNode] } = definition;
+
+const saveLocalizationsMock = jest.fn();
+const updateRouterMock = jest.fn();
+const getExitTranslationsMock = jest.fn();
+const getResultNameFieldMock = jest.fn();
+const onBindWidgetMock = jest.fn();
+
+const setup = composeSetup<GroupsRouterProps>(
+    {
+        translating: false,
+        localGroups: [],
+        nodeToEdit: groupsRouterNode,
+        saveLocalizations: saveLocalizationsMock,
+        updateRouter: updateRouterMock,
+        getExitTranslations: getExitTranslationsMock,
+        getResultNameField: getResultNameFieldMock,
+        onBindWidget: onBindWidgetMock
+    },
+    {
+        endpoints: config.endpoints
+    },
+    GroupsRouter
+);
+
+describe('GroupsRouter', () => {
+    describe('helpers', () => {
+        describe('extractGroups', () => {
+            it('should extract groups from the exits of a groupsRouter node', () => {
+                extractGroups(groupsRouterNode).forEach(({ name, id }, idx) => {
+                    expect(name).toBe(groupsRouterNode.exits[idx].name);
+                    expect(id).toBe(
+                        (groupsRouterNode.router as SwitchRouter).cases[idx].arguments[0]
+                    );
+                });
+            });
+        });
+
+        describe('hasGroupsRouter', () => {
+            it('should return true if given Node has a groups router', () => {
+                expect(hasGroupsRouter(groupsRouterNode)).toBeTruthy();
+            });
+
+            it('should return false if given NOde does not have a groups router', () => {
+                expect(hasGroupsRouter(sendMsgNode)).toBeFalsy();
+            });
+        });
+    });
+
+    describe('render', () => {
+        it('should render self, children', () => {
+            const { wrapper, props: { localGroups, nodeToEdit }, context: { endpoints } } = setup();
+            expect(getResultNameFieldMock).toHaveBeenCalledTimes(1);
+            expect(wrapper.find('.instructions').exists()).toBeTruthy();
+            expect(wrapper.find('p').text()).toBe(GROUP_LABEL);
+            expect(wrapper.find('GroupsElement').props()).toEqual({
```

Comment:
also, in that spirit, i think we should make it a rule to not use jest's `expect.objectContaining` assertion, which allows you to check for the presence of a subset of properties within an object. `placeholder` and `searchPromptText` are default props declared in `GroupsElement`, so i could have used it here to avoid bringing those constants in.